filters/CoordOperationFilter.cpp

```diff
@@ -0,0 +1,143 @@
+﻿/******************************************************************************
+* Copyright (c) 2019, Aurelien Vila (aurelien.vila@delair.aero)
+*
+* All rights reserved.
+*
+* Redistribution and use in source and binary forms, with or without
+* modification, are permitted provided that the following
+* conditions are met:
+*
+*     * Redistributions of source code must retain the above copyright
+*       notice, this list of conditions and the following disclaimer.
+*     * Redistributions in binary form must reproduce the above copyright
+*       notice, this list of conditions and the following disclaimer in
+*       the documentation and/or other materials provided
+*       with the distribution.
+*     * Neither the name of Hobu, Inc. or Flaxen Geo Consulting nor the
+*       names of its contributors may be used to endorse or promote
+*       products derived from this software without specific prior
+*       written permission.
+*
+* THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+* "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+* LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+* FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+* COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+* INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+* BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS
+* OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED
+* AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+* OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT
+* OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY
+* OF SUCH DAMAGE.
+****************************************************************************/
+
+#include "CoordOperationFilter.hpp"
+
+#include <gdal_version.h>
+#if GDAL_VERSION_MAJOR >= 3
+
+#include <pdal/PointView.hpp>
+#include <pdal/private/SrsTransform.hpp>
+#include <pdal/util/ProgramArgs.hpp>
+
+#include <ogr_spatialref.h>
+
+//#include <memory>
+
+namespace pdal
+{
+
+static StaticPluginInfo const s_info
+{
+    "filters.coordoperation",
+    "Transform coordinates using Proj pipeline string, WKT2 coordinate operations or URN definition",
```

Comment:
@vilaa Thanks for the contribution. It brings up some design and organization discussion the project needs to have in regards to this capability.

@abellgithub this filter really does exactly the same thing as `filters.projection` if we add the necessary capability to it. What value is there in having two filters with somewhat similar names doing slightly different things?